client/getipns.go

```diff
@@ -0,0 +1,70 @@
+package client
+
+import (
+	"context"
+
+	"github.com/ipfs/go-delegated-routing/gen/proto"
+)
+
+func (fp *Client) GetIPNS(ctx context.Context, id []byte) ([][]byte, error) {
+	resps, err := fp.GetIPNSAsync(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	records := [][]byte{}
+	for resp := range resps {
+		if resp.Err != nil {
+			records = append(records, resp.Record)
+		}
+	}
+	if fp.validator != nil {
```

Comment:
Can this ever happen? The validator is built in now so seems like we can remove this condition.